### DISCUSSION TITLE
https://theme-hope.vuejs.press/zh/guide/layout/sidebar.html

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#599] [Announcements]
# https://theme-hope.vuejs.press/zh/guide/layout/sidebar.html

侧边栏可能会包含相关的文档列表，文档标题以及博客模式下的博主信息。

https://theme-hope.vuejs.press/zh/guide/layout/sidebar.html

**Comment by siazon** [comment]
order: 1 不起作用，请问是在每个页面Frontmatter中设置这个参数吗？

**Reply by Mister-Hope** [reply]
![image](https://github.com/vuepress-theme-hope/giscus-discussions/assets/33315834/395888e4-07b1-41f0-9712-03a2bc318fe5)

**Comment by 1226085293** [comment]
对于文件夹的排序不起作用，目录结构
`
- 博客
  - A
  - B

`
我想要 B 在 A 的前面，我尝试在 A/B 的 README.md 设置 order 属性，但是它们不生效，有什么解决办法吗？

另外如果我在侧边栏配置中启用了 `children: "structure"`，如果想排除展示其中一个目录及子目录应该怎么配置，没找到相关的属性

**Reply by 1226085293** [reply]
注：A/B 都是文件夹

**Reply by Mister-Hope** [reply]
![image](https://github.com/vuepress-theme-hope/giscus-discussions/assets/33315834/6a73bbbe-3fd4-4097-a4e0-c04d5061f239)

**Comment by 137709772** [comment]
dir:
  order: 1
目录下的README.MD 用这个

**Reply by 137709772** [reply]
参照这个：
https://github.com/vuepress-theme-hope/vuepress-theme-hope/blob/main/docs/theme/src/zh/guide/layout/README.md?plain=1

**Comment by zenghr0820** [comment]
问一下我的文档使用了 permalink 属性后，侧边栏不显示，这个要怎么配置呢，有没有大佬知道的

**Reply by Mister-Hope** [reply]
如果你觉得它不工作，开一个issue。

侧边栏的分组配置是基于**实际网页路径**的，也就是如果你在 /a/ 文件夹下启用了结构化侧边栏，则只有访问 `/a/` 开头的路径才会拥有这个侧边栏，如果其中一个页面被设置了 `/test.html` 的永久路径，那么在 `/a/` 开头的任何路径下侧边栏应该包含它，当你访问 `/test.html` 时，此时会显示 `/` 的默认侧边栏，因为你已不在 `/a/` 下。在浏览器中主题并不清楚也并不关心这个页面的原有文件路径。

**Comment by yoyo-feifei** [comment]
除了启用、关闭，有收起侧边栏的设置吗？有个按钮点击展开侧边栏或者收起侧边栏。这样阅读正文时，可视空间还能更大些。

**Reply by Mister-Hope** [reply]
侧边栏是否可收起是基于屏幕宽度的，你可以自行调整断点。当屏幕足够宽时，我们认为无需隐藏侧边栏。如果你坚持，自己修改主题或使用其他主题或框架。

**Comment by phil-cheng** [comment]
当使用structure自动生成侧边栏时，在三级目录下的readme.md内设置index:false不生效，还是会把三级目录生成出来。

**Reply by Mister-Hope** [reply]
https://github.com/vuepress-theme-hope/giscus-discussions/discussions/599#discussioncomment-8703969

**Comment by ping-ter** [comment]
请问有办法默认收起侧边栏吗

**Reply by Mister-Hope** [reply]
https://github.com/vuepress-theme-hope/giscus-discussions/discussions/599#discussioncomment-8704021

**Comment by lumengde123** [comment]
缩小页面直到toc隐藏之后(平板端大小)，点击arrow start小图标可以切换显示/隐藏 侧边栏。我希望在PC页面也能使用这个功能，先是`display: flex`抵消掉了原本的`display: none`
```css
@media (min-width: 1440px) {
    .toggle-sidebar-wrapper {
        display: flex;
    }
}
```
把小图标显示出来了，但是点击没效果，可能width不够小触发不了后面的点击事件？不懂 T_T

**Reply by Mister-Hope** [reply]
你只能设置断点，把平板范围的断点扩大到足够大。

**Comment by w2534073922** [comment]
侧边栏设置了最大层级吗，我有很多层的目录但只展示了前几层

**Reply by w2534073922** [reply]
没问题，是我自己遍历目录的方式有问题。

**Comment by Zephery** [comment]
下面两种方式，展示的样式不一样啊？

```js
import { hopeTheme } from "vuepress-theme-hope";

export default hopeTheme({
  sidebar: [
    "/" /* / */,
    {
      text: "Foo",
      prefix: "/foo/",
      children: [
        "" /* /foo/ */,
        "one" /* /foo/one.html */,
        "two" /* /foo/two.html */,
      ],
    },
    {
      text: "Bar",
      prefix: "/bar/",
      children: [
        "" /* /bar/ */,
        "three" /* /bar/three.html */,
        "four" /* /bar/four.html */,
      ],
    },
    "/contact" /* /contact.html */,
    "/about" /* /about.html */,
  ],
});
```

```js
import { hopeTheme } from "vuepress-theme-hope";

export default hopeTheme({
  sidebar: {
    "/foo/": "structure",

    "/bar/": "structure",

    // fallback
    "/": ["" /* / */, "contact" /* /contact.html */, "about" /* /about.html */],
  },
});
```

**Comment by xiezhr** [comment]
测试一下
